src/test/javascript/spec/component/admin/legal-document-update.component.spec.ts

```diff
@@ -0,0 +1,154 @@
+        expect(component.unsavedChanges).toBeFalse();
+        expect(component.legalDocument).toEqual(returnValue);
+    });
+
+    it.each([LegalDocumentType.PRIVACY_STATEMENT, LegalDocumentType.IMPRINT])(
+        'should load legal document in selected language on language change',
+        (documentType: LegalDocumentType) => {
+            const returnValue = new LegalDocument(documentType, LegalDocumentLanguage.GERMAN);
+            returnValue.text = 'text';
+            setupRoutes(documentType);
+            component.ngOnInit();
+            let loadFile;
+            if (documentType === LegalDocumentType.PRIVACY_STATEMENT) {
+                loadFile = jest.spyOn(legalDocumentService, 'getPrivacyStatementForUpdate').mockReturnValue(of(returnValue));
+            } else {
+                loadFile = jest.spyOn(legalDocumentService, 'getImprintForUpdate').mockReturnValue(of(returnValue));
+            }
+            component.currentLanguage = LegalDocumentLanguage.GERMAN;
+            component.unsavedChanges = false;
+            component.onLanguageChange(LegalDocumentLanguage.ENGLISH);
+            fixture.detectChanges();
+            expect(loadFile).toHaveBeenCalledOnce();
+            expect(loadFile).toHaveBeenCalledWith(LegalDocumentLanguage.ENGLISH);
+            expect(component.currentLanguage).toEqual(LegalDocumentLanguage.ENGLISH);
+            expect(component.unsavedChanges).toBeFalse();
+        },
+    );
+
+    it('should correctly determine unsaved changes', () => {
+        component.unsavedChanges = false;
+        component.legalDocument.text = 'text';
+        component.checkUnsavedChanges('changed text');
+        expect(component.unsavedChanges).toBeTrue();
+    });
+
+    it.each([LegalDocumentType.PRIVACY_STATEMENT, LegalDocumentType.IMPRINT])(
+        'should call update legal document service method on update',
+        fakeAsync((documentType: LegalDocumentType) => {
+            const returnValue = new LegalDocument(documentType, LegalDocumentLanguage.GERMAN);
+            returnValue.text = 'text';
+            setupRoutes(documentType);
+            component.ngOnInit();
+            let updateFile;
+            if (documentType === LegalDocumentType.PRIVACY_STATEMENT) {
+                updateFile = jest.spyOn(legalDocumentService, 'updatePrivacyStatement').mockReturnValue(of(returnValue));
+            } else {
+                updateFile = jest.spyOn(legalDocumentService, 'updateImprint').mockReturnValue(of(returnValue));
+            }
+            component.markdownEditor.markdown = 'text';
+            component.unsavedChanges = true;
+            const expected = new LegalDocument(documentType, LegalDocumentLanguage.GERMAN);
+            expected.text = 'text';
+            fixture.detectChanges();
+            const button = fixture.nativeElement.querySelector('#update-legal-document-btn');
+            button.click();
+            tick(100);
```

Comment:
Why do you need to `tick(100)` here? Isn't a single tick usually enough?

Reply:
removed, not needed